config/sql/config.go

```diff
@@ -17,11 +17,38 @@ limitations under the License.
 package sql
 
 import (
+	"fmt"
+
 	"github.com/crossplane-contrib/terrajet/pkg/config"
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/pkg/errors"
 
 	"github.com/crossplane-contrib/provider-tf-azure/apis/rconfig"
 )
 
+func msSQLConnectionDetails(attr map[string]interface{}) (map[string][]byte, error) {
+	dbName, ok := attr["name"].(string)
+	if !ok {
+		return nil, errors.New("cannot get name attribute")
+	}
+	username, ok := attr["administrator_login"].(string)
+	if !ok {
+		return nil, errors.New("cannot get administrator_login attribute")
```

Comment:
These things depend on the authentication type, for example, it may not use username/password auth type. So, I'd suggest skipping instead of returning error for all connection detail keys if we can't find them in tfstate.

Reply:
Good point, done!

Reply:
I think we should treat `administrator_login` similar to `administrator_login_password` here.

Reply:
@muvaf not sure about this. Looks like `administrator_login` is a required attribute:
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/sql_server#administrator_login